pkgs/applications/editors/vscode/extensions/README.md

```diff
@@ -0,0 +1,14 @@
+# Visual Studio Code Extensions
+
+## Conventions for adding new extensions
+
+* Extensions are named in the **lowercase** version of the extension's unique identifier. Which is found on the marketplace extension page, and is the name under which the extension is installed by VSCode under `~/.vscode`.
+  Extension location should be: ${lib.strings.toLower mktplcRef.publisher}.${lib.string.toLower mktplcRef.name}
+
+* Move extension to a discrete directory whenever the extension needs `extra parameters` and other files (such as `patches` or `components`). The parameters that exist in the global index file are utilities shared by many extensions. Parameters that are specific to an extension or that are not globally used shouldn't be added to the global index page.
+
+* Currently `nixfmt-rfc-style` formatter is being used to format the VSCode extensions.
```

Comment:
I don't know much about `nixfmt-rfc-style`, is this a new standard formatter being used everywhere, or is this just something you're using? If there isn't consensus on this in general. I'd probably drop it.

Reply:
https://github.com/NixOS/rfcs/pull/166